From 1b8a639bfdce84c9b39cd9e89b6da4c1d06cc7ab Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Sun, 4 Feb 2024 19:40:32 +0800
Subject: [PATCH] policy/modules/system/systemd: allow systemd-logind to
 inherit local login file descriptors

Fix reboot timeout error:
$ reboot
Failed to set wall message, ignoring: Failed to activate service 'org.freedesktop.login1': timed out (service_start_timeout=25000ms)
Call to Reboot failed: Failed to activate service 'org.freedesktop.login1': timed out (service_start_timeout=25000ms)

avc:  denied  { use } for  pid=287 comm="systemd-logind"
path="anon_inode:[pidfd]" dev="anon_inodefs" ino=1044
scontext=system_u:system_r:systemd_logind_t
tcontext=system_u:system_r:local_login_t tclass=fd permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index ffce3c0e8..03aeb8515 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -973,6 +973,7 @@ init_stop_system(systemd_logind_t)
 miscfiles_read_localization(systemd_logind_t)
 
 locallogin_read_state(systemd_logind_t)
+locallogin_use_fds(systemd_logind_t)
 
 seutil_libselinux_linked(systemd_logind_t)
 seutil_read_default_contexts(systemd_logind_t)
-- 
2.25.1

